specifyweb/frontend/js_src/lib/components/Formatters/Formatter.tsx

```diff
@@ -145,51 +147,142 @@ function Definitions({
     }
   }, [needFormatter, handleChange, formatter.definition.fields]);
 
+  const [showConditionalField, setShowConditionalField] = React.useState(
+    trimmedFields.map(() => false)
+  );
+
   return table === undefined ? null : (
     <div className="flex flex-col gap-4 divide-y divide-gray-500 [&>*]:pt-4">
       {trimmedFields.map(({ value, fields }, index) => (
-        <div className="flex flex-col gap-2" key={index}>
+        <div
+          className={`flex ${
+            showConditionalField[index] || !hasCondition
+              ? 'flex-col'
+              : 'items-center'
+          } gap-2`}
+          key={index}
+        >
           {hasCondition && (
             <Label.Block>
-              {resourcesText.conditionFieldValue()}
-              <Input.Text
-                isReadOnly={isReadOnly}
-                value={value ?? ''}
-                onValueChange={(value): void =>
-                  handleChanged(
-                    {
-                      value: value.length === 0 ? undefined : value,
-                      fields,
-                    },
-                    index
-                  )
-                }
-              />
-              <span>
-                {index === 0
-                  ? resourcesText.elseConditionDescription()
-                  : resourcesText.conditionDescription()}
-              </span>
+              {showConditionalField[index]
+                ? resourcesText.conditionFieldValue()
+                : null}
+              <div className="flex items-center gap-2">
+                <Input.Text
+                  className="h-full"
+                  isReadOnly={isReadOnly}
+                  value={value ?? ''}
+                  onValueChange={(value): void =>
+                    handleChanged(
+                      {
+                        value: value.length === 0 ? undefined : value,
+                        fields,
+                      },
+                      index
+                    )
+                  }
+                />
+                {trimmedFields.length > 0 && showConditionalField[index] ? (
+                  <Button.Danger
+                    className="w-[40%]"
```

Comment:
the red button seems too large

![Screenshot 2024-01-10 at 19 16 49](https://github.com/specify/specify7/assets/40512816/d062e14a-3b2b-4390-aca7-fdd5cf9d64f6)

can we remove className?

```suggestion
```

Reply:
If we remove the class name it looks like this:
![Screenshot 2024-01-11 at 10 00 42 AM](https://github.com/specify/specify7/assets/108160931/d3904087-ddf2-4945-ad49-bc85bf570abd)

Reply:
I see. Can you please add one of these class names instead of hardcoding the size:
- https://tailwindcss.com/docs/text-wrap
- https://tailwindcss.com/docs/word-break
- https://tailwindcss.com/docs/whitespace

alternatively, this might display correctly if you wrap either the Input or the Button in a `<div>`